packages/typespec-client-generator-core/src/package.ts

```diff
@@ -291,6 +291,9 @@ function getPropertyPathFromSegment(
   for (const segment of segments) {
     const property = current.properties.get(segment);
     if (!property) {
+      if (current.baseModel) {
+        return getPropertyPathFromSegment(context, current.baseModel, segments);
+      }
```

Comment:
Should we do a `while` (if baseModel has another baseModel)?

Reply:
if the base does not have the property we want and also has base, it will recursive call `getPropertyPathFromSegment`.